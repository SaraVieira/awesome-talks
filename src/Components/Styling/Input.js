import styled from 'styled-components'
import remcalc from 'remcalc'

export default styled.input`
    font-size: ${remcalc(15)};
    color: ${props => props.theme.tertiary};
    width: 100%;
    box-sizing: border-box;
    letter-spacing: ${remcalc(1)};
    border: 0;
    padding: ${remcalc(7)} 0;
    border-bottom: ${remcalc(1)} solid #ccc;

    &:focus {
        outline: none;
    }

    & ~ span {
        position: absolute;
        bottom: 0;
        left: 0;
        width: 0;
        height: ${remcalc(2)};
        background-color: #337294;
        transition: 0.4s;
    }
    &:focus ~ span {
        width: 100%;
        transition: 0.4s;
    }
`
