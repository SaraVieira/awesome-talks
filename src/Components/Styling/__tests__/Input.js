import React from 'react'
import remcalc from 'remcalc'
import Shallow from '../../../Tests/shallow'
import Input from '../Input'

test('Renders Input', () => {
    const Wrapper = Shallow(<Input />)

    expect(Wrapper).toHaveStyleRule('border-bottom', `${remcalc(1)} solid #ccc`)
    expect(Wrapper).toHaveStyleRule('font-size', remcalc(15))
    expect(Wrapper).toMatchSnapshot()
})
