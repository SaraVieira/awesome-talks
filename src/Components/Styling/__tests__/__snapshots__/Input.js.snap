// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Input 1`] = `
.c0 {
  font-size: 0.9375rem;
  width: 100%;
  box-sizing: border-box;
  -webkit-letter-spacing: 0.0625rem;
  -moz-letter-spacing: 0.0625rem;
  -ms-letter-spacing: 0.0625rem;
  letter-spacing: 0.0625rem;
  border: 0;
  padding: 0.4375rem 0;
  border-bottom: 0.0625rem solid #ccc;
}

.c0:focus {
  outline: none;
}

.c0 ~ span {
  position: absolute;
  bottom: 0;
  left: 0;
  width: 0;
  height: 0.125rem;
  background-color: #337294;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

.c0:focus ~ span {
  width: 100%;
  -webkit-transition: 0.4s;
  transition: 0.4s;
}

<input
  className="c0"
/>
`;
